patterns/state/ships/src/main/java/com/scottlogic/dp/state/ships/ShipState.java

```diff
@@ -1,21 +1,17 @@
 package com.scottlogic.dp.state.ships;
 
 abstract class ShipState {
-    abstract void startFishing(FishingShip ship);
+    abstract ShipState startFishing();
 
-    void stopFishing(FishingShip ship) {
-        System.out.println("Haven't thrown that net yet, mate.");
-    };
+    abstract ShipState stopFishing();
```

Comment:
Just a note, it seems that the this class you're specifying methods as package-private but the implementing classes have them public :)